tango/asyncio_executor.py

```diff
@@ -94,7 +90,7 @@ def submit(self, fn, *args, **kwargs):
 
     def execute(self, fn, *args, **kwargs):
         """Execute an operation and return the result."""
-        if self.loop._thread_id == get_ident():
+        if self.in_executor_context:
```

Comment:
Isn't self.in_executor_context a method returning a boolean?

Reply:
It's actually a property, but you're right, it doesn't really make the code any cleaner. I'll change that.